proposals/0000-add-support-for-display-p3-colors.md

```diff
@@ -0,0 +1,202 @@
+---
+title: Add Support for Wide Gamut (DisplayP3) Colors to React Native
+author:
+  - Ryan Linton
+  - Yulian Glukhenko
+date: 2023-11-20
+---
+
+# RFC0000: Add Support for Wide Gamut (DisplayP3) Colors to React Native
+
+## Summary
+
+React Native does [not currently support](https://github.com/facebook/react-native/issues/41517) wide gamut color spaces (i.e. display-p3). This proposal discusses adding support for this color space in React Native, covering:
+
+- JS implementation
+- Platform changes
+
+## Basic example
+
+Add a DisplayP3 background color and text color using color() function syntax per the [W3C CSS Color Module Level 4](https://www.w3.org/TR/css-color-4/#color-function) spec.
+
+### Using StyleSheet
+
+```js
+StyleSheet.create({
+  view: { backgroundColor: "color(display-p3 1 0.5 0)" },
+  text: { color: "color(display-p3 0 0.5 1)" },
+});
+```
+
+### Using inline styles
+
+```jsx
+const MyComp = () => (
+  <View style={{ backgroundColor: "color(display-p3 1 0.5 0)" }}>
+    <Text style={{ color: "color(display-p3 0 0.5 1)"  }}>
+  </View>
+)
+```
+
+## Motivation
+
+Since most new devices support wider gamut color spaces, React Native should support them as well. The DisplayP3 color space has had native support since Android 8.0 and iOS 9.3. The `color()` function was introduced with [CSS Color Module Level 4](https://www.w3.org/TR/css-color-4/#color-function), much of which is already implemented in React Native. Flutter also [recently added support](https://github.com/flutter/flutter/issues/55092) for DisplayP3 on iOS with plans to follow up with support for Android and then the framework itself. Support for DisplayP3 colors can improve the quality of React Native apps by providing developers a wider range of available colors to use and is aligned with React Native's mission to provide the same look and feel as the underlying native platforms.
+
+## Detailed design
+
+### JS Changes
+
+1. Parse the color() function in [normalizeColor](https://github.com/facebook/react-native/blob/63213712125795ac082597dad2716258b90cdcd5/packages/normalize-color/index.js#L235)
+
+```js
```

Comment:
The motivation and API here make sense. How close is this towards a PR?

Reply:
I haven't done any implementation work yet. Riccardo has been helping me with this so we were planning on leaving this open for feedback for a couple weeks before starting on implementation when he gets back.